fees/hono.ts

```diff
@@ -0,0 +1,67 @@
+import { Adapter, FetchResultFees } from "../adapters/types";
+import { CHAIN} from "../helpers/chains";
+import { request, gql } from "graphql-request";
+import type { ChainEndpoints } from "../adapters/types"
+import { Chain } from '@defillama/sdk/build/general';
+import { getPrices } from "../utils/prices";
+
+
+interface IData {
+  id: string;
+  todayETHRevenue: string;
+}
+
+const endpoints = {
+  [CHAIN.ETHEREUM]: "https://api.thegraph.com/subgraphs/name/nuoanunu/defilahma-revenue-hono"
+};
+
+const graph = (graphUrls: ChainEndpoints) => {
+  const graphQuery = gql`query fees($timestampFrom: Int!, $timestampTo: Int!)
+  {
+    dailyRevenueAggregators(first:1, where:{id_gte:$timestampFrom, id_lte:$timestampTo})
+    {
+      id
+      todayETHRevenue
+    }
+  }`;
+
+  return (chain: Chain) => {
+    return async (timestamp: number): Promise<FetchResultFees> => {
+
+      const fromTimestamp = timestamp - 60 * 60 * 24
+      const toTimestamp = timestamp
+
+      const graphRes: IData[] = (await request(graphUrls[chain], graphQuery, {
+        timestampFrom: fromTimestamp,
+        timestampTo: toTimestamp
+      })).dailyRevenueAggregators;
+
+      const prices = await getPrices(["ethereum:0x0000000000000000000000000000000000000000"], timestamp);
+      try {
+        const dailyRevenue = (Number(graphRes[0].todayETHRevenue))/10**18 * prices["ethereum:0x0000000000000000000000000000000000000000"].price;
+      return {
+        dailyRevenue: `${dailyRevenue}`,
+        timestamp
+      }
+      } catch (error) {
+    
+      }
+      return {
```

Comment:
many this is should inside `catch` ?

Reply:
how may bad catch should throw error like this `throw error`

Reply:
There is only 1 case: where no revenue is found on that day. As we operate like a hedge fund, not everyday there will be revenue. 
*For example when the market is bad, no trading volumne => almost 0 to no revenue => Meaning then the revenue collector wont be active => no events => no data